client/state/data-layer/wpcom/videos/poster/index.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { VIDEO_EDITOR_UPDATE_POSTER } from 'state/action-types';
+import {
+	setPosterUrl,
+	showError,
+	showUploadProgress,
+} from 'state/ui/editor/video-editor/actions';
+
+/**
+ * Updates the poster for a video.
+ *
+ * @param  {Object} store  Redux store
+ * @param  {Object} action  Action object
+ * @param {Function} next  Dispatches to next middleware in chain
+ * @returns {Object} original action
+ */
+export const updatePoster = ( { dispatch }, action, next ) => {
+	if ( ! ( 'file' in action.params ) && ! ( 'at_time' in action.params ) ) {
+		return next( action );
+	}
+
+	const params = {
+		apiVersion: '1.1',
+		method: 'POST',
+		path: `/videos/${ action.videoId }/poster`,
+	};
+
+	if ( 'file' in action.params ) {
+		params.formData = [ [ 'poster', action.params.file ] ];
+	}
+
+	if ( 'at_time' in action.params ) {
+		params.body = action.params;
+	}
+
+	dispatch( http( params, action ) );
+
+	return next( action );
+};
+
+export const updatePosterUrl = ( { dispatch }, action, next, { poster } ) => {
+	dispatch( setPosterUrl( poster ) );
+};
+
+export const updatePosterError = ( { dispatch } ) => {
+	dispatch( showError() );
+};
+
+export const updateUploadProgress = ( { dispatch }, action, next, progress ) => {
+	let percentage = 0;
+
+	if ( 'loaded' in progress && 'total' in progress ) {
```

Comment:
also @donnapep I think we may have some choices in those cases where `total` might be something like `Infinity` if `lengthComputable` is `false`. that should still indicate as much where progress is concerned and it will give consumers the ability to check if they need to